server/src/main/scala/viper/gobraserver/ServerConfig.scala

```diff
@@ -0,0 +1,74 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobraserver
+
+import org.rogach.scallop.{ScallopConf, ScallopOption, intConverter, numberHandler, singleArgConverter}
+import viper.server.core.DefaultVerificationExecutionContext
+
+class ServerConfig(arguments: Seq[String])
+  extends ScallopConf(arguments) {
+
+  /**
+    * Prologue
+    */
+
+  version(
+    s"""
+       | ${Server.name} ${Server.copyright}
+       |   version ${Server.version}
+     """.stripMargin
+  )
+
+  banner(
+    s""" Usage: ${Server.name} [OPTIONS]
+       |
+       | Options:
+       |""".stripMargin
+  )
+
+  /**
+    * Command-line options
+    */
+  private val portOpt: ScallopOption[Int] = opt[Int](
+    name = "port",
+    descr = s"Port on which ${Server.name} should listen. Port '0' will result in choosing an arbitrary available port.",
+    default = Some(0)
+  )
+
+  /**
+    * minimum number of threads that Gobra Server needs; it's one more than the minimum number of threads for the
+    * default verification execution context as 1 thread will be used to by LSP4j to process the sockets
+    */
+  val minNumberOfThreads: Int = DefaultVerificationExecutionContext.minNumberOfThreads + 1
```

Comment:
Did I see right that the Gobra IDE, except the LSP4j thread, never spawns another thread?

Reply:
I would say that this is true conceptually. Note however, that several operations on futures require/use the execution context (e.g. `transform`). Thus, they implicitly submit additional jobs to the thread pool.
However, I've empirically evaluated the above minimum of threads and your observation is correct as these implicit jobs by futures run after another job from Gobra or ViperServer and thus do not require an additional thread to not deadlock.

Reply:
It is all fine, I somehow missed the futures